### PR TITLE
Calculate reserve auction price using kicked amount

### PR DESCRIPTION
**Pull request by EdNoepel**
## Description

Contradicting the whitepaper, Claimable Reserve Auctions (CRAs) originally had a pricing mechanism with a start price of 1 quote token for 1bbn AJNA.  This was done to simplify the auction mechanism, but the whitepaper was not updated to reflect the implementation.  Kirill's audit brought this up with issue M-01.  Upon initial review we favored the original implementation and chose to update the whitepaper.  This decision was challenged during the fix review.

The original implementation naturally prevented 0-bids, as the pricing function would not reach 0 after 72 hours.  Since the amount of claimable reserves is now part of the formula, we can no longer guarantee this.  As such, an explicit check has been added to ensure at least 1 wei of AJNA is burned.  This change necessitated improvement in logic handling bids for less than the quote token's trading increment.

To reduce cost of updating unit tests, `RewardsManager` tests are now skipped.  Removal of `RewardsManager` may be covered by a separate PR.

Note that the author is not necessarily in agreement with this change.

## Purpose

(versus updating whitepaper)

### Pros
- CRAs will reach a fair market price more quickly

### Cons
- additional gas cost for kicking, exascerbated by removal of the kick reward in this release
- added maintenance complexity
- CRAs will reach an infinitessimal price where they may be almost-zero-bid upon more quickly

## Impact

Pool contract sizes are similar.
2.3x gas cost for kicking, due to writing to an additional storage slot.  Trivial increase in gas cost for taking.

## Tasks

- [x] Changes to protocol contracts are covered by unit tests executed by CI.
- [x] Protocol contract size limits have not been exceeded.
- [x] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [x] Scope labels have been assigned as appropriate.
- [ ] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

File: tests/forge/unit/ERC20Pool/ERC20PoolReserveAuction.t.sol [hunk 1]
```diff
@@ -113,7 +113,7 @@ contract ERC20PoolReserveAuctionTest is ERC20HelperContract {
         _pool.takeReserves(10 * 1e18);
         assertEq(USDC.balanceOf(address(_pool)),   1_006.397978 * 1e6);
         assertEq(USDC.balanceOf(address(_bidder)), 1.471235 * 1e6);
-        assertEq(AJNA.balanceOf(address(_bidder)), 9.999999998723906366 * 1e18);
+        assertEq(AJNA.balanceOf(address(_bidder)), 9.999999999132638263 * 1e18);
     }
 
     function testZeroBid() external {
```
Comment (ith-harvey) [inline tests/forge/unit/ERC20Pool/ERC20PoolReserveAuction.t.sol]: does this method need `tearDown`?

**Review by ith-harvey — APPROVED** [review]
LGTM

File: src/libraries/helpers/PoolHelper.sol [hunk 2]
```diff
@@ -382,17 +382,20 @@ import { Maths }   from '../internal/Maths.sol';
...
+        uint256 lastKickedReserves_
     ) view returns (uint256 price_) {
         if (reserveAuctionKicked_ != 0) {
             uint256 secondsElapsed   = block.timestamp - reserveAuctionKicked_;
             uint256 hoursComponent   = 1e27 >> secondsElapsed / 3600;
             uint256 minutesComponent = Maths.rpow(MINUTE_HALF_LIFE, secondsElapsed % 3600 / 60);
+            uint256 initialPrice     = lastKickedReserves_ == 0 ? 0 : Maths.wdiv(1_000_000_000 * 1e18, lastKickedReserves_);
 
-            price_ = Maths.rayToWad(1_000_000_000 * Maths.rmul(hoursComponent, minutesComponent));
+            price_ = initialPrice * Maths.rmul(hoursComponent, minutesComponent) / 1e27;
```
Comment (ith-harvey) [inline src/libraries/helpers/PoolHelper.sol:398]: What is the reason we are no longer using `Maths.rayToWad()` here?

Comment (EdNoepel) [reply src/libraries/helpers/PoolHelper.sol:398]: We're now multiplying a WAD by a RAY and scaling back down to a WAD.  Was trying to save resources by not creating a one-off _Math.wrmulw_ library method for this.

File: tests/forge/unit/ERC20Pool/ERC20PoolReserveAuction.t.sol (hunk 1 above)
Comment (EdNoepel) [reply tests/forge/unit/ERC20Pool/ERC20PoolReserveAuction.t.sol]: Sure; added in 5a5b7a3a7ef338b68a0207673036f4992dd59eed.

**Review by mattcushman — APPROVED** [review]
lgtm

**Review by k1rill-fedoseev — APPROVED** [review]
LGTM

`rayToWad` seems like is no longer used, so can be removed, shouldn't affect the bytecode size though.

**Review by grandizzy — APPROVED** [review]
Lgtm, would be nice to remove `rayToWad` too as Kirill pointed out

**Comment by ith-harvey** [comment]
> LGTM
> 
> `rayToWad` seems like is no longer used, so can be removed, shouldn't affect the bytecode size though.

done -> https://github.com/ajna-finance/contracts/pull/1032/commits/f9c22d57c4725b3b2022eb9aca5254e55a475b47

**Review by dmitriia — COMMENTED** [review]
Looks ok

**Pull request merged**